crates/network/src/lib.rs

```diff
@@ -0,0 +1,218 @@
+use {
+    ethcontract::{
+        jsonrpc::serde::{de, Deserialize, Deserializer},
+        U256,
+    },
+    thiserror::Error,
+};
+
+/// Represents each available network
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum Network {
+    Mainnet = 1,
+    Goerli = 5,
+    Gnosis = 100,
+    Sepolia = 11155111,
+    ArbitrumOne = 42161,
+    Base = 8453,
+}
+
+impl Network {
+    /// Returns the network's chain ID
+    pub fn chain_id(&self) -> u64 {
+        match self {
+            Self::Mainnet => 1,
+            Self::Goerli => 5,
+            Self::Gnosis => 100,
+            Self::Sepolia => 11155111,
+            Self::ArbitrumOne => 42161,
+            Self::Base => 8453,
+        }
+    }
+
+    /// Returns the canonical name of the network on CoW Protocol.
+    pub fn name(&self) -> &'static str {
+        // You can find a list of available networks by network and chain id here:
+        // https://chainid.network/chains.json
+        match &self {
+            Self::Mainnet => "Ethereum / Mainnet",
+            Self::Goerli => "Ethereum / Goerli",
+            Self::Gnosis => "xDAI",
+            Self::Sepolia => "Ethereum / Sepolia",
+            Self::ArbitrumOne => "Arbitrum One",
+            Self::Base => "Base",
+        }
+    }
+
+    pub fn default_amount_to_estimate_native_prices_with(&self) -> Option<U256> {
+        match &self {
+            Self::Mainnet | Self::Goerli | Self::Sepolia | Self::ArbitrumOne | Self::Base => {
+                Some(10u128.pow(18).into())
+            }
+            Self::Gnosis => Some(10u128.pow(21).into()),
+        }
+    }
+
+    /// Returns the block time in milliseconds
+    pub fn block_time_in_ms(&self) -> u64 {
+        match self {
+            Self::Mainnet => 13_000,
```

Comment:
```suggestion
            Self::Mainnet => 12_000,
```

Reply:
No, not too important from the aspect of settlement code where it was used as 13_000